java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -200,7 +200,7 @@ public void writeTo(DataWriter dout) throws IOException {
 
       // Header for each column...
       for (int i = 0; i < numColumns; i++) {
-        dout.writeInt(types[i].nativeId);
+        dout.writeInt(types[i].getNativeId());
         dout.writeInt((int) nullCounts[i]);
```

Comment:
We need to update the code here to deal with scale too.

Reply:
done